core/src/main/scala/org/apache/spark/Dependency.scala

```diff
@@ -122,6 +120,14 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
    */
   private[this] var _shuffleMergedFinalized: Boolean = false
 
+  /**
+   * shuffleMergeId is used to give temporal ordering to the executions of a ShuffleDependency.
+   * This is required in order to handle indeterministic stage retries for push-based shuffle.
+   */
+  private[this] var _shuffleMergeId: Int = -1
```

Comment:
start with `0`

Reply:
Now `shuffleMergeId` starts with 0 and also for indeterminate stages the first time it is submitted we are calling `newShuffleMergeState` so that would increment it to `1` and so for all indeterminate stages the `shuffleMergeId` starts from 1.